internal/tracker/oci.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package tracker
+
+import (
+	"context"
+	"io"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/static"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+const (
+	unknownConfig       = "application/vnd.unknown.config.v1+json"
+	openPolicyAgentData = "application/vnd.cncf.openpolicyagent.data.layer.v1+json"
+	title               = "org.opencontainers.image.title"
+	dataFileTitle       = "data/data/acceptable_tekton_bundles.yml"
+)
+
+func PullImage(ctx context.Context, imageRef string) ([]byte, error) {
+	ref, err := name.ParseReference(imageRef)
+	if err != nil {
+		return nil, err
+	}
+
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		return nil, err
+	}
+
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	var data []byte
+	for i, layer := range layers {
+		mediaType, err := layer.MediaType()
+		if err != nil {
+			return nil, err
+		}
+
+		if mediaType != openPolicyAgentData {
+			continue
+		}
+
+		if i > 0 {
```

Comment:
If there are other layers with different unexpected mediaTypes, you could end up with something like this:

```yaml
foo: bar
---
---
spam: maps
```

I think that should be ok because the code should handle an empty doc properly.

Reply:
I could cover that case, but I don't think we'll have more than one layer regardless (this should be fine:tm:)